src/arrayMethodSort.js

```diff
@@ -3,9 +3,34 @@
 /**
  * Implement method Sort
  */
+const defaultCompare = function (a, b) {
+  const aStr = String(a);
+  const bStr = String(b);
+
+  if (aStr < bStr) {
+    return -1;
+  }
+
+  if (aStr > bStr) {
+    return 1;
+  }
+
+  if (aStr === bStr) {
+    return 0;
+  }
+};
+
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction = defaultCompare) {
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < this.length - 1; j++) {
+        if (compareFunction(this[j], this[j + 1]) > 0) {
+          [this[j], this[j + 1]] = [this[j + 1], this[j]];
+        }
+      }
+    }
```

Comment:
While the bubble sort algorithm is implemented correctly, it is not the most efficient sorting algorithm for larger arrays. Consider using more advanced algorithms like quicksort or mergesort for better performance on larger datasets.